main.cpp

```diff
@@ -269,6 +176,20 @@ struct FloatType
 {
     FloatType(float floatIn_) : value(new float(floatIn_)){}
     
+    FloatType(const FloatType& other)
+    {
+        value = new float();
+        *value = *other.value;
+    }
+    
+    FloatType& operator= (const FloatType& other)
+    {
+        delete value;
+        value = new float();
+        *value = *other.value;
+        return *this;
+    }
```

Comment:
You do not need these Rule of 3 functions, and they are implemented incorrectly.  You have a leak!
- [ ] remove these